commit a958f2d82735e177df442b577b197f96305edbc6
Author: Hans Rosenfeld <hans.rosenfeld@joyent.com>
Date:   2019-09-11T16:01:09+02:00 (4 weeks ago)
    
    OS-7981 mpt_sas hangs after config header request timeout
    Reviewed by: Robert Mustacchi <rm+illumos@fingolfin.org>
    Reviewed by: Jerry Jelinek <jerry.jelinek@joyent.com>
    Approved by: Jerry Jelinek <jerry.jelinek@joyent.com>

diff --git a/usr/src/uts/common/io/scsi/adapters/mpt_sas/mptsas_impl.c b/usr/src/uts/common/io/scsi/adapters/mpt_sas/mptsas_impl.c
index 87abb767ad..c4a856b2bf 100644
--- a/usr/src/uts/common/io/scsi/adapters/mpt_sas/mptsas_impl.c
+++ b/usr/src/uts/common/io/scsi/adapters/mpt_sas/mptsas_impl.c
@@ -24,7 +24,7 @@
  * Copyright 2012 Nexenta Systems, Inc. All rights reserved.
  * Copyright 2014 OmniTI Computer Consulting, Inc. All rights reserved.
  * Copyright (c) 2014, Tegile Systems Inc. All rights reserved.
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright (c) 2019, Joyent, Inc.
  */
 
 /*
@@ -378,6 +378,7 @@ mptsas_access_config_page(mptsas_t *mpt, uint8_t action, uint8_t page_type,
 	 * Check if the header request completed without timing out
 	 */
 	if (cmd->cmd_flags & CFLAG_TIMEOUT) {
+		config_flags |= MPTSAS_CMD_TIMEOUT;
 		mptsas_log(mpt, CE_WARN, "config header request timeout");
 		rval = DDI_FAILURE;
 		goto page_done;
@@ -520,6 +521,7 @@ mptsas_access_config_page(mptsas_t *mpt, uint8_t action, uint8_t page_type,
 	 * Check if the request completed without timing out
 	 */
 	if (cmd->cmd_flags & CFLAG_TIMEOUT) {
+		config_flags |= MPTSAS_CMD_TIMEOUT;
 		mptsas_log(mpt, CE_WARN, "config page request timeout");
 		rval = DDI_FAILURE;
 		goto page_done;
